packages/transaction-controller/src/TransactionController.ts

```diff
@@ -411,26 +480,25 @@ export class TransactionController extends BaseControllerV1<
       getCurrentNetworkEIP1559Compatibility: () => Promise<boolean>;
       getExternalPendingTransactions?: (
         address: string,
+        chainId?: string,
       ) => NonceTrackerTransaction[];
       getGasFeeEstimates?: () => Promise<GasFeeState>;
       getNetworkState: () => NetworkState;
       getPermittedAccounts: (origin?: string) => Promise<string[]>;
       getSavedGasFees?: (chainId: Hex) => SavedGasFees | undefined;
       getSelectedAddress: () => string;
-      incomingTransactions?: {
-        includeTokenTransfers?: boolean;
-        isEnabled?: () => boolean;
-        queryEntireHistory?: boolean;
-        updateTransactions?: boolean;
-      };
+      incomingTransactions?: IncomingTransactionOptions;
       messenger: TransactionControllerMessenger;
       onNetworkStateChange: (listener: (state: NetworkState) => void) => void;
-      pendingTransactions?: {
-        isResubmitEnabled?: boolean;
-      };
+      pendingTransactions?: PendingTransactionOptions;
       provider: Provider;
       securityProviderRequest?: SecurityProviderRequest;
       speedUpMultiplier?: number;
+      findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+      getNetworkClientById: NetworkController['getNetworkClientById'];
+      getNetworkClientRegistry: NetworkController['getNetworkClientRegistry'];
+      hub: TransactionControllerEventEmitter;
```

Comment:
Am I right this is just to support the integration tests?

In the past I've just mocked the `EventEmitter` constructor via Jest and then you have total freedom to mock the entire object or just inject a instance you can `emit` with.

Could that be suitable here to avoid the test only constructor options?

Reply:
ill take a look at this

Reply:
@matthewwalsh0 do you have an example/link of the `EventEmitter` mocking you are talking about.

Reply:
https://github.com/MetaMask/core/blob/main/packages/transaction-controller/src/TransactionController.test.ts#L634

In this example we're mocking the `PendingTransactionTracker` constructor in the test, so we can control a mock instance to isolate the unit tests to the target logic rather than a separate unit with its own tests.

Reply:
The hub argument is removed in this cleanup PR: https://github.com/MetaMask/core/pull/3885